tests/sentry/api/test_event_search.py

```diff
@@ -768,271 +768,917 @@ def test_empty_string(self):
+            InvalidSearchQuery,
+            "Project {} does not exist or is not an actively selected project.".format(
+                project3.slug
+            ),
+        ):
+            get_filter(
+                "project:{} OR project:{}".format(project1.slug, project3.slug),
+                params={
+                    "organization_id": self.organization.id,
+                    "project_id": [project1.id, project2.id],
+                },
+            )
+
+    def test_issue_id_alias_in_condition_filters(self):
+        def _eq(xy):
+            return ["equals", [["ifNull", [xy[0], "''"]], xy[1]]]
+
+        group1 = self.create_group(project=self.project)
+        group2 = self.create_group(project=self.project)
+        group3 = self.create_group(project=self.project)
+        tests = [
+            (
+                "issue.id:{} OR issue.id:{}".format(group1.id, group2.id),
+                [],
+                [group1.id, group2.id],
+            ),
+            ("issue.id:{} AND issue.id:{}".format(group1.id, group1.id), [], [group1.id]),
+            (
+                "(issue.id:{} AND issue.id:{}) OR issue.id:{}".format(
+                    group1.id, group2.id, group3.id
+                ),
+                [],
+                [group1.id, group2.id, group3.id],
+            ),
+            ("issue.id:{} OR a:b".format(group1.id), [[_eq("ab"), "=", 1]], [group1.id]),
+            ("issue.id:{} AND a:b".format(group1.id), [[_eq("ab"), "=", 1]], [group1.id]),
+            (
+                "(issue.id:{} AND a:b) OR issue.id:{}".format(group1.id, group2.id),
+                [[_eq("ab"), "=", 1]],
+                [group1.id, group2.id],
+            ),
+            (
+                "(issue.id:{} AND a:b) OR c:d".format(group1.id),
+                [[["or", [_eq("ab"), _eq("cd")]], "=", 1]],
+                [group1.id],
+            ),
+        ]
+
+        for test in tests:
+            result = get_filter(
+                test[0],
+                params={"organization_id": self.organization.id, "project_id": [self.project.id]},
+            )
+            assert test[1] == result.conditions, test[0]
+            assert test[2] == result.group_ids, test[0]
+
+    def test_invalid_conditional_filters(self):
+        with self.assertRaisesRegexp(
+            InvalidSearchQuery, "Condition is missing on the left side of 'OR' operator"
+        ):
+            get_filter("OR a:b")
+
+        with self.assertRaisesRegexp(
+            InvalidSearchQuery, "Missing condition in between two condition operators: 'OR AND'"
+        ):
+            get_filter("a:b Or And c:d")
+
+        with self.assertRaisesRegexp(
+            InvalidSearchQuery, "Condition is missing on the right side of 'AND' operator"
+        ):
+            get_filter("a:b AND c:d AND")
+
+        with self.assertRaisesRegexp(
+            InvalidSearchQuery, "Condition is missing on the left side of 'OR' operator"
```

Comment:
Would be super :rainbow: if we could tell them the column offset of the problem as well in the future.